File: features/application-sections/profile-section/warehouse-feature/warehouse-impl/src/androidMain/kotlin/com/profile/Barcode.java
package com.profile;

import android.os.Parcel;
import android.os.Parcelable;

public class Barcode implements Parcelable {
    private byte[] barcodeData;
    private int barcodeType;
    private int fromScannerID;

    public Barcode(byte[] barcodeData, int barcodeType, int fromScannerID) {
        this.barcodeData = barcodeData;
        this.barcodeType = barcodeType;
        this.fromScannerID = fromScannerID;
    }

    protected Barcode(Parcel in) {
        this.barcodeData = in.createByteArray();
        this.barcodeType = in.readInt();
        this.fromScannerID = in.readInt();
    }

    public byte[] getBarcodeData() {
        return barcodeData;
    }

    public void setBarcodeData(byte[] barcodeData) {
        this.barcodeData = barcodeData;
    }

    public int getBarcodeType() {
        return barcodeType;
    }

    public void setBarcodeType(int barcodeType) {
        this.barcodeType = barcodeType;
    }

    public int getFromScannerID() {
        return fromScannerID;
    }

    public void setFromScannerID(int fromScannerID) {
        this.fromScannerID = fromScannerID;
    }

    @Override
    public void writeToParcel(Parcel parcel, int flags) {
        parcel.writeByteArray(barcodeData);
        parcel.writeInt(barcodeType);
        parcel.writeInt(fromScannerID);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Barcode> CREATOR = new Creator<Barcode>() {
        @Override
        public Barcode createFromParcel(Parcel in) {
            return new Barcode(in);
        }

        @Override
        public Barcode[] newArray(int size) {
            return new Barcode[size];
        }
    };
}

File: features/application-sections/profile-section/warehouse-feature/warehouse-impl/src/androidMain/kotlin/com/profile/profile/screens/main_refactor/screens/arrival_and_consumption_goods/Barcode.kt
package com.profile.profile.screens.main_refactor.screens.arrival_and_consumption_goods

import android.os.Parcel
import android.os.Parcelable

class Barcode : Parcelable {
    var barcodeData: ByteArray
    var barcodeType: Int
    var fromScannerID: Int

    constructor(barcodeData: ByteArray, barcodeType: Int, fromScannerID: Int) {
        this.barcodeData = barcodeData
        this.barcodeType = barcodeType
        this.fromScannerID = fromScannerID
    }

    constructor(`in`: Parcel) {
        barcodeData = `in`.readString()!!.toByteArray()
        barcodeType = `in`.readInt()
        fromScannerID = `in`.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(String(barcodeData))
        parcel.writeInt(barcodeType)
        parcel.writeInt(fromScannerID)
    }

    companion object CREATOR : Parcelable.Creator<Barcode> {
        override fun createFromParcel(parcel: Parcel): Barcode {
            return Barcode(parcel)
        }

        override fun newArray(size: Int): Array<Barcode?> {
            return arrayOfNulls(size)
        }
    }
}